### PR TITLE
Added gcode file type association

### PR DESCRIPTION
**Pull request by Ghostkeeper**
This adds file type association for g-code files to open with Cura.

### CODE REVIEW

**Comment by Ghostkeeper** [comment]
The following files are still missing file type associations with g-codes. Some of them are really out of sync.

* The `cura.sharedmimeinfo` file in the Cura repository, used for Debian packaging.
* The `cura.desktop.in` file in the Cura repository.

Also the MIME type for g-code appears to have settled on `text/x.gcode`. Source: https://bugs.freedesktop.org/show_bug.cgi?id=99421 (see the bottommost reactions in that thread).

File: cmake/NSIS.template.in [hunk 1]
```diff
@@ -557,6 +557,10 @@ Section /o "Open X3D files with Cura"
     ${RegisterExtension} "$INSTDIR\Cura.exe" ".x3d" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.x3d.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@" "X3D model file"
 SectionEnd
 
+Section /o "Open GCODE files with Cura"
+    ${RegisterExtension} "$INSTDIR\Cura.exe" ".gcode" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.gcode.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@" "GCODE model file"
```
Comment (Ghostkeeper) [inline cmake/NSIS.template.in]: We also support the ".g" extension.

File: cmake/NSIS.template.in [hunk 2]
```diff
@@ -876,11 +881,12 @@ Section "Uninstall"
   Delete "$INSTDIR\AddRemove.exe"
 !endif
 
-  ;Unregister file assiciations
+  ;Unregister file associations
   ${UnRegisterExtension} ".3mf" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.3mf.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@"
   ${UnRegisterExtension} ".obj" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.obj.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@"
   ${UnRegisterExtension} ".stl" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.stl.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@"
   ${UnRegisterExtension} ".x3d" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.x3d.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@"
+  ${UnRegisterExtension} ".gcode" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.gcode.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@"
```
Comment (nallath) [inline cmake/NSIS.template.in]:
```suggestion
  ${UnRegisterExtension} ".gcode" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.gcode.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@"
    ${UnRegisterExtension} ".g" "@CPACK_PACKAGE_NAME_NO_WHITESPACES@.gcode.@CPACK_PACKAGE_VERSION_MAJOR@.@CPACK_PACKAGE_VERSION_MINOR@"
```

Comment (LipuFei) [reply cmake/NSIS.template.in]: Added the counterpart in dbb12dd

**Pull request merged**